builtin/providers/aws/resource_aws_cloudwatch_logs_subscription_filter.go

```diff
@@ -0,0 +1,341 @@
+		FilterPattern:  aws.String(filter_pattern),
+		LogGroupName:   aws.String(log_group),
+	}
+
+	if _, ok := d.GetOk("role"); ok {
+		params.RoleArn = aws.String(d.Get("role").(string))
+	}
+
+	return params
+}
+
+func resourceAwsCloudwatchLogsSubscriptionFilterRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudwatchlogsconn
+
+	log_group := d.Get("log_group").(string)
+	name := d.Get("name").(string) // "name" is a required field in the schema
+
+	req := &cloudwatchlogs.DescribeSubscriptionFiltersInput{
+		LogGroupName:     aws.String(log_group),
+		FilterNamePrefix: aws.String(name),
+	}
+
+	resp, err := conn.DescribeSubscriptionFilters(req)
+	if err != nil {
+		return fmt.Errorf("Error reading SubscriptionFilters for log group %s with name prefix %s: %#v", log_group, d.Get("name").(string), err)
+	}
+
+	for _, subscriptionFilter := range resp.SubscriptionFilters {
+		if *subscriptionFilter.LogGroupName == log_group {
+			d.SetId(cloudwatchLogsSubscriptionFilterId(log_group))
+			return nil // OK, matching subscription filter found
+		}
+	}
+
+	return fmt.Errorf("Subscription filter for log group %s with name prefix %s not found!", log_group, d.Get("name").(string))
+}
+
+func resourceAwsCloudwatchLogsSubscriptionFilterDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudwatchlogsconn
+
+	log_group := d.Get("log_group").(string)
+	name := d.Get("name").(string)
+	destination := d.Get("destination").(string)
+
+	if strings.HasPrefix(destination, "arn:aws:lambda") {
+		// access permissions should also be cleaned up
+		lambda_conn := meta.(*AWSClient).lambdaconn
+
+		lambda_arn_sliced := strings.Split(destination, ":")
+		function_name := lambda_arn_sliced[len(lambda_arn_sliced)-1]
+		statement_id := lambdaPermissionStatementId(log_group, destination)
+
+		if permissionExists(function_name, statement_id, lambda_conn) {
+			_, err := lambda_conn.RemovePermission(&lambda.RemovePermissionInput{
+				FunctionName: aws.String(function_name),
+				StatementId:  aws.String(statement_id),
+			})
+			if err != nil {
+				if awsErr, ok := err.(awserr.Error); ok {
+					log.Printf("[WARN] Error removing the access permission SID (%s) for lambda function (%s), message: \"%s\", code: \"%s\"",
+						statement_id, function_name, awsErr.Message(), awsErr.Code())
+				} else {
+					log.Printf("[WARN] Error removing the access permission from lambda function %s: %#v", function_name, err)
+				}
+			}
+		}
+	}
+
+	params := &cloudwatchlogs.DeleteSubscriptionFilterInput{
+		FilterName:   aws.String(name),      // Required
+		LogGroupName: aws.String(log_group), // Required
+	}
+	_, err := conn.DeleteSubscriptionFilter(params)
+
+	if err != nil {
+		return fmt.Errorf(
+			"Error deleting Subscription Filter from log group: %s with name filter name %s", log_group, name)
+	}
+	d.SetId("")
+	return nil
+}
+
+func waitForKinesisStreamToActivate(conn *kinesis.Kinesis, stream_name string) error {
```

Comment:
This shouldn't be necessary if the Kinesis stream is managed by Terraform:
https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_kinesis_stream.go#L61-L68

I'm not sure whether the user would expect an error if they provide a stream that is not ACTIVE yet... personally I probably would.